[2024-01-31T15:19:31.340+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: stock_analysis_to_gcs_dag.process_and_save_data_to_gcs manual__2024-01-31T15:19:23.224918+00:00 [queued]>
[2024-01-31T15:19:31.344+0000] {taskinstance.py:1957} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: stock_analysis_to_gcs_dag.process_and_save_data_to_gcs manual__2024-01-31T15:19:23.224918+00:00 [queued]>
[2024-01-31T15:19:31.344+0000] {taskinstance.py:2171} INFO - Starting attempt 1 of 2
[2024-01-31T15:19:31.354+0000] {taskinstance.py:2192} INFO - Executing <Task(PythonOperator): process_and_save_data_to_gcs> on 2024-01-31 15:19:23.224918+00:00
[2024-01-31T15:19:31.357+0000] {standard_task_runner.py:60} INFO - Started process 6243 to run task
[2024-01-31T15:19:31.359+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'stock_analysis_to_gcs_dag', 'process_and_save_data_to_gcs', 'manual__2024-01-31T15:19:23.224918+00:00', '--job-id', '89', '--raw', '--subdir', 'DAGS_FOLDER/stock_analysis_to_gcs_dag.py', '--cfg-path', '/tmp/tmpzqnvc8q9']
[2024-01-31T15:19:31.361+0000] {standard_task_runner.py:88} INFO - Job 89: Subtask process_and_save_data_to_gcs
[2024-01-31T15:19:31.382+0000] {task_command.py:423} INFO - Running <TaskInstance: stock_analysis_to_gcs_dag.process_and_save_data_to_gcs manual__2024-01-31T15:19:23.224918+00:00 [running]> on host de-airflow-vm-instance-1.europe-west10-a.c.dataengineeringfinal-411911.internal
[2024-01-31T15:19:31.423+0000] {taskinstance.py:2481} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='stock_analysis_to_gcs_dag' AIRFLOW_CTX_TASK_ID='process_and_save_data_to_gcs' AIRFLOW_CTX_EXECUTION_DATE='2024-01-31T15:19:23.224918+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-01-31T15:19:23.224918+00:00'
[2024-01-31T15:19:31.428+0000] {logging_mixin.py:188} INFO - ------------------DATA-------------: {'Information': 'Thank you for using Alpha Vantage! Our standard API rate limit is 25 requests per day. Please subscribe to any of the premium plans at https://www.alphavantage.co/premium/ to instantly remove all daily rate limits.'}
[2024-01-31T15:19:31.429+0000] {taskinstance.py:2699} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/deprojectue/miniconda3/envs/airflow_demo/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/deprojectue/miniconda3/envs/airflow_demo/lib/python3.9/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/deprojectue/miniconda3/envs/airflow_demo/lib/python3.9/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/home/deprojectue/airflow_demo/dags/stock_analysis_to_gcs_dag.py", line 19, in process_and_save_to_gcs
    df = pd.DataFrame.from_dict(stock_data["Time Series (Daily)"], orient='index')
KeyError: 'Time Series (Daily)'
[2024-01-31T15:19:31.431+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=stock_analysis_to_gcs_dag, task_id=process_and_save_data_to_gcs, execution_date=20240131T151923, start_date=20240131T151931, end_date=20240131T151931
[2024-01-31T15:19:31.440+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 89 for task process_and_save_data_to_gcs ('Time Series (Daily)'; 6243)
[2024-01-31T15:19:31.450+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-01-31T15:19:31.458+0000] {taskinstance.py:3281} INFO - 0 downstream tasks scheduled from follow-on schedule check
